core/primitives/src/shard_layout.rs

```diff
@@ -241,9 +241,34 @@ pub enum ShardLayoutError {
 }
 
 impl ShardLayout {
-    /* Some constructors */
-    pub fn v0_single_shard() -> Self {
-        Self::v0(1, 0)
+    /// Construct a layout with a single shard
+    pub fn single_shard() -> Self {
+        Self::of_num_shards(1, 0)
+    }
+
+    /// Construct a layout with given number of shards
+    pub fn of_num_shards(num_shards: NumShards, version: ShardVersion) -> Self {
```

Comment:
got a better idea for the fn name?

Reply:
Maybe `multi_shard`, just to mach the `single_shard` one?

Reply:
That was my first thought but it could also be used to create a single-shard layout, so I changed my mind. But if you like that name I'm also down with it. :)